src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -792,35 +870,53 @@ bool PDPServer2::process_changes_release()
                 // everyone, time at which it is removed from history.
                 if (ch->kind == fastrtps::rtps::ChangeKind_t::ALIVE)
                 {
-                    remove_change_from_writer_history(mp_PDPWriter, mp_PDPWriterHistory, ch);
+                    if (!remove_change_from_writer_history(mp_PDPWriter, mp_PDPWriterHistory, ch))
```

Comment:
Comment on why if the change could not be removed from the writer history it should be released manually to return the change to the pool.